### DISCUSSION TITLE
Home Assistant Core -- From Python 3.11.2 to Python 3.12.x

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#2458] [Announcements]
1) Begin by creating a new Home Assistant Core LXC.

2) Next, to transfer your data to the new LXC, execute the following command within the Proxmox shell:

```bash
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/copy-data/home-assistant-core-copy-data-home-assistant-core.sh)"
```
(From: https://github.com/tteck/Proxmox/blob/main/misc/copy-data/README.md)

3) Once the data transfer is complete, shut down your original Home Assistant Core LXC. Transfer the MAC address to the new Home Assistant Core LXC and start it. Done

If you wish to retain the same CTID, follow these additional steps:

- After completing the above steps, remove the old Home Assistant Core LXC to free up its CTID.
- Make a backup of the new Home Assistant Core LXC.
- Navigate to the storage location where the backup is stored and select it for restoration, but do not choose "Start after restore". This will automatically assign the first available CTID, which should match your old Home Assistant Core LXC CTID.
- Shut down and remove the Home Assistant Core LXC with the undesired CTID, then start the Home Assistant Core LXC with the desired CTID.

**Comment by tteck** [comment]
Expected Python Versions:

- 3.12.2: Tuesday, 2024-02-06
- 3.12.3: Tuesday, 2024-04-09
- 3.12.4: Tuesday, 2024-06-04
- 3.12.5: Tuesday, 2024-08-06
- 3.12.6: Tuesday, 2024-10-01
- 3.12.7: Tuesday, 2024-12-03
- 3.12.8: Tuesday, 2025-02-04
- 3.12.9: Tuesday, 2025-04-08

![Screenshot 2024-02-08 8 42 09 PM](https://github.com/tteck/Proxmox/assets/21342670/91005c6c-c09f-4f9e-9fb3-db80b8fa869a)

**Comment by Tux92** [comment]
The script works, all settings is imported well, but database is corrupted and homeassistant create a new one. All my historical data gone.

So I remain with old lxc for now.

**Comment by Tux92** [comment]
Update: the procedure is ok. Remind only to shutdown source LXC before use posted script otherwise homeassistant db will be corrupted.

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/commit/57d06ad2bef3e2dc01a4ba16297d830096203d2c

**Comment by gcortes** [comment]
In addition to the MAC address, make sure to copy over pertinent setting. I had set up my original home assistant on a different bridge and forgot to do so in the new version.

If you're also running Node-red, you may have to restart it. Mine was getting connection errors after the switch. Restarting the service fixed it.

**Comment by julred** [comment]
The above mentioned workaround doesnt work for me. I shut down the old home assistant lxc.
I installed a new instance of Home Assistant - started your script from PVE shell and copied everything.
Aferward it was not possible to start home assistant (ssh was available ).
The new lxc container ip container was visible in my router - but webinterface does not respond - no webinterface available.
Any recommondation / things I can try to solve?

**Reply by julred** [reply]
@tteck Any suggestions?

**Reply by tteck** [reply]
I have no idea why this is happening to you.

**Reply by elad-bar** [reply]
ha cannot run smoothly with python3.11, it throws errors, to solve it, it should have dependancy of python3.12 (At least),
happens to me with clean install of new LXC container
